Error when executing make

I get:

    Could not open requirements file: [Errno 2] No such file or directory: 'requirements/edx/github.txt'
    make: *** [lms_install_github_prereqs] Error 1

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,15 @@
+sudo: required
+
+language: python
+
+services:
+  - docker
+
+cache: pip
+
+before_install:
+  - pip install docker-compose==1.5.1
+
+script:
+  - make
+  - docker-compose start

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ all: build_stuff
 	echo Now you can run the stack using $ docker-compose up
 
 
-build_stuff: build_images lms_install_github_prereqs db_initialize_mysql
+build_stuff: build_images db_initialize_mysql
 	# Build the images.
 	echo Done buidling images and database
 
@@ -48,11 +48,6 @@ build_images:
 	# This step will take some time to build the image...
 	docker-compose build
 
-
-lms_install_github_prereqs:
-	# Installs the github requirements in the `edx-platform/src/` dir.
-
-	docker-compose run lms pip install --exists-action w -r requirements/edx/github.txt
 
 db_initialize_mysql:
 	# Build the database from scratch: create database and migrate it.

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -2,17 +2,16 @@ mysql:
   image: mysql:5.5
   environment:
     MYSQL_ROOT_PASSWORD: "password"
-  volumes:
-   - "./docker-vols/mysql_lib:/var/lib/mysql"
-   - "./docker-vols/mysql_tmp_cache:/tmp/cache"
+  volumes_from:
+    - "data:rw"
 
 memcached:
   image: memcached:1.4
 
 mongo:
   image: mongo:2.4
-  volumes:
-    - "./docker-vols/mongo_data_db:/data/db"
+  volumes_from:
+    - "data:rw"
 
 lms:
   build: edxapp
@@ -20,23 +19,11 @@ lms:
     PYTHONPATH: "/edx/app/edxapp/src/"
     DJANGO_SETTINGS_MODULE: lms.envs.docker
 
-  # command: paver lms
-  command: python manage.py lms runserver 0.0.0.0:8000
+  command: paver lms
+  # command: python manage.py lms runserver 0.0.0.0:8000
 
-  volumes:
-   - "/edx/app/edxapp/edx-platform:/edx/app/edxapp/edx-platform"
-
-   - "./docker-vols/lms_edx_app_edxapp_edx_platform_src:/edx/app/edxapp/edx-platform/src/"
-   - "./docker-vols/lms_edx_app_edxapp_staticfiles:/edx/app/edxapp/staticfiles/"
-
-   - "./docker-vols/lms_edx_app_edxapp_log:/edx/app/edxapp/log"
-   - "./docker-vols/lms_edx_var_edxapp_data:/edx/var/edxapp/data"
-   - "./docker-vols/lms_edx_app_edxapp_uploads:/edx/app/edxapp/uploads"
-
-   - "./edxapp/lms_docker_env.py:/edx/app/edxapp/edx-platform/lms/envs/docker.py"
-   - "./edxapp/cms_docker_env.py:/edx/app/edxapp/edx-platform/cms/envs/docker.py"
-   - "./edxapp/docker_common.py:/edx/app/edxapp/edx-platform/lms/envs/docker_common.py"
-   - "./edxapp/docker_common.py:/edx/app/edxapp/edx-platform/cms/envs/docker_common.py"
+  volumes_from:
+    - "data:rw"
 
   links:
    - mysql
@@ -49,16 +36,32 @@ studio:
     PYTHONPATH: "/edx/app/edxapp/src/"
     DJANGO_SETTINGS_MODULE: cms.envs.docker
 
-  # command: paver studio
-  command: python manage.py cms runserver 0.0.0.0:8001
+  command: paver studio
+  # command: python manage.py cms runserver 0.0.0.0:8001
 
   volumes_from:
-    - lms:rw
+    - data:rw
 
   links:
    - mysql
    - mongo
    - memcached
+
+data:
+  image: busybox
+  volumes:
+    # MySQL
+    - /var/lib/mysql
+    - /tmp/cache
+
+    # MongoDB
+    - /data/db
+
+    # edxapp
+    - /edx/app/edxapp/staticfiles/
+    - /edx/app/edxapp/log
+    - /edx/var/edxapp/data
+    - /edx/app/edxapp/uploads
 
 nginx:
   build: nginx

--- a/edxapp/Dockerfile
+++ b/edxapp/Dockerfile
@@ -3,6 +3,7 @@ FROM ubuntu:14.04
 MAINTAINER Omar Al-Ithawi <i@omardo.com>
 
 # The latest from edx/master that worked with me
+# change it to the commit you'd like
 ENV EDX_VERSION 7a858b6e09afd96e68939e4074825d1459144a2e
 
 # Use: git references to speed up clone
@@ -11,13 +12,18 @@ ENV EDX_REPO https://github.com/edx/edx-platform.git
 RUN mkdir -p /edx/app/edxapp/
 WORKDIR /edx/app/edxapp/
 
-RUN apt-get update -y
-# TODO: Use apt-get cache to speedup install
-RUN apt-get install software-properties-common git -y
+RUN apt-get update -y  \
+    && apt-get install -y  \
+            git  \
+            python-pip  \
+            ruby1.9.3  \
+            software-properties-common
 
 RUN git clone $EDX_REPO
 WORKDIR /edx/app/edxapp/edx-platform
 RUN git checkout $EDX_VERSION
+
+RUN gem install bundler
 
 # TODO: Add this using the apt-repos.txt file
 RUN apt-add-repository ppa:chris-lea/node.js
@@ -27,23 +33,22 @@ RUN apt-get update -y
 # RUN rm /etc/apt/apt.conf.d/no-cache
 ADD apt-packages.gitpatch /tmp/
 RUN git apply /tmp/apt-packages.gitpatch
-# TODO: Use apt-get cache to speedup install
 RUN cat requirements/system/ubuntu/apt-packages.txt | xargs apt-get install -y
 
-# TODO: Put on `apt-packages.txt` file
-# TODO: Use apt-get cache to speedup install
-RUN apt-get install python-pip -y
 
-# TODO: Use both `pip` and `git` cahce to speed installs
-# TODO: Avoid this step and run it and `make` time
-RUN for pckg in pre base post paver development github local; \
-      do pip install --exists-action w -r requirements/edx/$pckg.txt; \
-    done
+# Keep `pre.txt` in its own step to support options in base and other stuff
+RUN pip install --exists-action w -r requirements/edx/pre.txt
 
+# TODO: Use the following to reduce layers
+    # for pckg in $REQUIREMENT_FILES; \
+    #   do pip install -q --disable-pip-version-check --exists-action w -r requirements/edx/$pckg.txt; \
+    # done
+RUN pip install -q --disable-pip-version-check --exists-action w -r requirements/edx/github.txt
+RUN pip install -q --disable-pip-version-check --exists-action w -r requirements/edx/local.txt
+RUN pip install -q --disable-pip-version-check --exists-action w -r requirements/edx/base.txt
+RUN pip install -q --disable-pip-version-check --exists-action w -r requirements/edx/post.txt
+RUN pip install -q --disable-pip-version-check --exists-action w -r requirements/edx/paver.txt
 
-# TODO: Use apt-get cache to speedup install
-RUN apt-get install ruby1.9.3 -y
-RUN gem install bundler
 RUN bundle install
 
 # Fix hardcoded values
@@ -51,9 +56,6 @@ RUN mkdir -p /edx/app/edxapp/log \
     && mkdir -p /edx/var/edxapp/data \
     && mkdir -p /edx/app/edxapp/uploads
 
-
-# TODO: Fix it in the actual code
-# ADD logsettings.py openedx/core/lib/
 
 # Keep those at last to have less cache invalidations
 
@@ -67,9 +69,6 @@ ADD docker_common.py lms/envs/
 ADD docker_common.py cms/envs/
 
 RUN paver install_prereqs
-
-# TODO: Change the `apt-packages.gitpatch` instead of having this step
-RUN apt-get -y install yui-compressor
 
 RUN mkdir /edx/app/edxapp/staticfiles/
 

